File: payroll-api/src/main/java/com/company/payroll/model/StaffAttendance.java
package com.company.payroll.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@RequiredArgsConstructor
public class StaffAttendance {
    private Integer atId;
    private LocalDate attendanceDate;
    private LocalDateTime checkInOne;
    private LocalDateTime checkOutOne;
    private LocalDateTime checkInTwo;
    private LocalDateTime checkOutTwo;
    private LocalDateTime checkInThree;
    private LocalDateTime checkOutThree;
    private Integer staffId;
    private LocalDate startDate;
    private LocalDate endDate;
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/model/StaffAttendance.kt
package com.company.payroll.model

import com.company.payroll.annotation.NoArg
import java.time.LocalDate
import java.time.LocalDateTime

/**
 * @author caroline
 *
 * <p> Created at 24 June 2023
 * <p> Use as model for table staff_attendance
 */
@NoArg
data class StaffAttendance(
    var atId: Int,
    var attendanceDate: LocalDate?,
    var checkInOne: LocalDateTime?,
    var checkOutOne: LocalDateTime?,
    var checkInTwo: LocalDateTime?,
    var checkOutTwo: LocalDateTime?,
    var checkInThree: LocalDateTime?,
    var checkOutThree: LocalDateTime?,
    var staffId: Int,
    var startDate: LocalDate?,
    var endDate: LocalDate?
)
